llama-index-integrations/embeddings/llama-index-embeddings-voyageai/llama_index/embeddings/voyageai/base.py

```diff
@@ -43,58 +50,60 @@ def __init__(
             **kwargs,
         )
 
+        self.client = voyageai.Client(api_key=voyage_api_key)
+
     @classmethod
     def class_name(cls) -> str:
         return "VoyageEmbedding"
 
     def _get_query_embedding(self, query: str) -> List[float]:
         """Get query embedding."""
-        return self._model.get_embedding(
-            query, model=self.model_name, input_type="query"
-        )
+        return self.client.embed(
+            [query], model=self.model_name, input_type="query"
+        ).embeddings[0]
 
     async def _aget_query_embedding(self, query: str) -> List[float]:
         """The asynchronous version of _get_query_embedding."""
-        return await self._model.aget_embedding(
-            query, model=self.model_name, input_type="query"
-        )
+        return await self.client.embed(
```

Comment:
I realized that this is not the correct way to do async calls. We should use an async client. Please see https://github.com/voyage-ai/voyageai-python/blob/main/voyageai/client_async.py
It is currently not included in the docs, but will be included soon.

Reply:
No, it is not the proper way. Actually, i checked the client.py, but did not find the async implementation, thanks for sharing this implementation, i'm going to use the same!

Also, the same mistake is present in langchain voyageai package, i'll use the async client there as well.

Reply:
I was wondering if this could be implemented differently. The async function could be implemented in the Client as a separate function - so the consumer can call the sync or the async function from the same client instance.